Automatically detect hostname

cweiske opened the pull request:
.. but also allow manual hostname configuration.

I had to add a commit to fix the dev mode build.

Resolves: https://github.com/ActivityWatch/aw-watcher-web/issues/132
Obsoletes: https://github.com/ActivityWatch/aw-watcher-web/pull/143
Obsoletes: https://github.com/kepstin/aw-watcher-web/pull/1
<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Automatically detects hostname with manual override option, updating client initialization, settings UI, and storage handling.
> 
>   - **Behavior**:
>     - Automatically detects hostname using `detectHostname()` in `client.ts`.
>     - Allows manual hostname configuration via settings UI in `index.html` and `main.ts` in `settings`.
>     - Updates `getBucketId()` in `client.ts` to include hostname in bucket ID if available.
>   - **UI Changes**:
>     - Adds hostname input field to `settings/index.html`.
>     - Replaces edit button with settings button in `popup/index.html` and `popup/main.ts`.
>   - **Storage**:
>     - Adds `getHostname()` and `setHostname()` in `storage.ts` for hostname management.
>   - **Misc**:
>     - Updates `vite.config.ts` to include settings files in build process.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=ActivityWatch%2Faw-watcher-web&utm_source=github&utm_medium=referral)<sup> for 7158b9225e144d794696c3d020d074fcb14cfe24. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

cweiske commented on src/background/client.ts:
```diff
@@ -24,13 +24,29 @@ export function ensureBucket(client: AWClient, bucketId: string) {
+    .then((info) => {
+      return info.hostname
+    })
+    .catch((err) => {
```
undefined hostname is handled in the code.

cweiske commented on src/settings/main.ts:
```diff
@@ -22,6 +22,9 @@ async function saveOptions(e: SubmitEvent): Promise<void> {
     selectedBrowser = customBrowserInput.value.toLowerCase()
   }
 
+  const hostnameInput = document.querySelector<HTMLInputElement>('#hostname')
```
it's always there.

BelKed commented:
@ellipsis-dev please review :)

BelKed commented on src/settings/index.html:
Please, undo the formatting/format the file correctly (`make format`) :)

cweiske replied:
done.

BelKed commented on vite.config.ts:
```diff
@@ -20,7 +20,10 @@ export default defineConfig({
     webExtension({
       manifest: generateManifest,
-      additionalInputs: ['src/consent/index.html', 'src/consent/main.ts'],
+      additionalInputs: [
+          'src/consent/index.html', 'src/consent/main.ts',
+          'src/settings/index.html', 'src/settings/main.ts'
+      ],
```
The change won't be needed anymore, it's fixed in the `master` now :)

BelKed reviewed (changes requested):
<p align=center>
<img width="70%" alt="Screenshot 2025-02-11 at 18 55 06" src="https://github.com/user-attachments/assets/f042483c-1108-441b-829a-d4d34af2d89a" />
</p>

I'd prefer to have the **Hostname** in same style as **Browser** (add `:` and make it bold), and under the Browser selection, because it might get quite wide once user selects other browser.

Also, move the **Save** button under **Hostname**, please :)

---

<p align=center>
<img width="70%" alt="Screenshot 2025-02-11 at 18 55 14" src="https://github.com/user-attachments/assets/5e7b4007-2655-4a6f-ab38-3ce6311567c2" />
</p>

Maybe we could also show the **Hostname** here? (Take inspiration from **Browser**)

BelKed reviewed (changes requested):
It lets user save with blank hostname, please make the hostname field required :)

BelKed commented:
Manually tested, works as intended :)

---

If the user was using the extension before, they will now have two buckets (one without the hostname and one with the correct hostname). This doesn't cause a problem, it gets merged correctly by the web UI :D
@ErikBjare what do you think about this? :)

BelKed commented:
It tries to create a bucket without hostname?
Now testing on Chrome, and it can't create the bucket...
After restart of Chrome it can, weird...
(It goes to the correct bucket with hostname now)

---

<p align=center>

<img width="70%" alt="Screenshot 2025-02-11 at 19 39 49" src="https://github.com/user-attachments/assets/3b73439c-0652-43bb-b006-3df5f0a15176" />

</p>

---

From `aw-server`:
```
2025-02-11 19:33:37 [ERROR]: Exception on /api/0/buckets/aw-watcher-web-chrome [POST]  (aw-server:1414)
Traceback (most recent call last):
  File "flask/app.py", line 1484, in full_dispatch_request
  File "flask/app.py", line 1469, in dispatch_request
  File "aw_server/rest.py", line 45, in decorator
  File "flask_restx/api.py", line 404, in wrapper
  File "flask/views.py", line 109, in view
  File "flask_restx/resource.py", line 46, in dispatch_request
  File "aw_server/rest.py", line 153, in post
KeyError: 'hostname'
```

cweiske commented:
I've rebased against master, reformatted and modified the settings window (hostname required and bold).
I have no idea how to get a newline into the settings window. Neither `<br/>` nor `display: block` helps.
I would not put it into the popup because it doesn't help. If you want to see the hostname, look into the settings - otherwise the user doesn't need to be bothered about it. (I personally also would remove the browser setting from the popup for the same reason).

If you want more changes, please add additional commits yourself.
